contracts/WildcardSteward_v3.sol

```diff
@@ -576,24 +620,31 @@ contract WildcardSteward_v3 is Initializable {
         (transferSuccess, ) = recipient.call.gas(2300).value(_wei)("");
     }
 
+    // if credit balance exists,
+    // if amount owed > creidt
+    // credit zero add amount
+    // else reduce credit by certain amount.
+    // else if credit balance doesn't exist
+    // add amount to balance
+
     function _updateBenefactorBalance(address benefactor) public {
-        uint256 unclaimedPayoutAvailable = patronageDueBenefactor(benefactor);
+        uint256 patronageDueBenefactor = patronageDueBenefactor(benefactor);
 
-        if (unclaimedPayoutAvailable > 0) {
-            if (
-                unclaimedPayoutAvailable.add(benefactorFunds[benefactor]) <
-                benefactorCredit[benefactor]
-            ) {
-                // Here there is nothing left extra to pay the organisation, everything goes to paying of debt.
-                benefactorCredit[benefactor] = benefactorCredit[benefactor].sub(
-                    unclaimedPayoutAvailable.add(benefactorFunds[benefactor])
-                );
+        if (patronageDueBenefactor > 0) {
+            if (benefactorCredit[benefactor] > 0) {
```

Comment:
I do prefer the way you are doing it (either the `benefactorCredit` or the `benefactorFunds` are null), but it unfortunately means more things to be careful of - because the compiler won't catch if this requirement ever isn't true; and if it isn't true our code breaks.

My idea would be to never change these values directly, and rather have a helper function called "addToBenefactorFunds" and "subtractFromBenefactorFunds", and those functions are the single points to enforce this condition.

Just a thought.

Reply:
Get what you saying. 100% we have to be careful about it.

Reply:
Let me write a yung function that does this quick...